react-carousel-component/src/carousel.jsx

```diff
@@ -0,0 +1,138 @@
+import React from 'react';
+
+const characters = [
+  {
+    id: 1,
+    name: 'Webber',
+    url: 'images/webber.jpg'
+  },
+  {
+    id: 2,
+    name: 'Wilson',
+    url: 'images/wilson.jpg'
+  },
+  {
+    id: 3,
+    name: 'Willow',
+    url: 'images/willow.jpg'
+  },
+  {
+    id: 4,
+    name: 'Wendy',
+    url: 'images/wendy.jpg'
+  },
+  {
+    id: 5,
+    name: 'Woody',
+    url: 'images/woody.jpg'
+  }
+];
+
+class Carousel extends React.Component {
+  constructor(props) {
+    super(props);
+    this.state = {
+      activeImgId: 1,
+      characters: characters
+    };
+    this.handleClick = this.handleClick.bind(this);
+    this.autoSwitchImg = this.autoSwitchImg.bind(this);
+  }
+
+  componentDidMount() {
+    setInterval(this.autoSwitchImg, 3000);
+  }
+
+  autoSwitchImg() {
+    if (this.state.activeImgId === characters.length) {
+      this.setState({
+        activeImgId: 1
+      });
+    } else {
+      this.setState({
+        activeImgId: this.state.activeImgId + 1
+      });
+    }
+  }
+
+  handleClick(event) {
+    if ((this.state.activeImgId === characters.length) && (event.target.getAttribute('class') === 'fa-solid fa-angle-right')) {
+      this.setState({
+        activeImgId: 1
+      });
+    } else if (event.target.getAttribute('class') === 'fa-solid fa-angle-right') {
+      this.setState({
+        activeImgId: this.state.activeImgId + 1
+      });
+    } else if ((this.state.activeImgId === 1) && (event.target.getAttribute('class') === 'fa-solid fa-angle-left')) {
+      this.setState({
+        activeImgId: 5
+      });
+    } else if (event.target.getAttribute('class') === 'fa-solid fa-angle-left') {
```

Comment:
This `5` seems too specific. Try to leverage the length of your array as much as possible to avoid requiring a specific number of images being passed to the component.